wagtailcache/cache.py

```diff
@@ -339,25 +342,15 @@ def clear_cache(urls: List[str] = []) -> None:
         return
 
     _wagcache = caches[wagtailcache_settings.WAGTAIL_CACHE_BACKEND]
-    if urls and "keyring" in _wagcache:
-        keyring = _wagcache.get("keyring")
-        # Check the provided URL matches a key in our keyring.
-        matched_urls = []
-        for regex in urls:
-            for key in keyring:
-                if re.match(regex, key):
-                    matched_urls.append(key)
-        # If it matches, delete each entry from the cache,
-        # and delete the URL from the keyring.
-        for url in matched_urls:
-            entries = keyring.get(url, [])
-            for cache_key in entries:
-                _wagcache.delete(cache_key)
-            del keyring[url]
-        # Save the keyring.
-        _wagcache.set("keyring", keyring)
+    if urls:
+        active_keys = KeyringItem.objects.active_for_url_regexes(
```

Comment:
If this returns a querystring... could you not instead call `KeyringItem.objects.active_for_url_regexes(urls).delete()` ?